scripts/download_artifacts.py

```diff
@@ -0,0 +1,151 @@
+    @param artifact_name is the artifact associated with build success
+    If the artifact does not exist, something failed and logs error into
+    appropriate file; else returns artifact id for downloading
+    """
+    build_name = artifact_name
+    build_failed = False
+    artifact = repo.get_artifacts(artifact_name).get_page(0)
+    # check build failed
+    if len(artifact) == 0:
+        print(f"build failed for {build_name}")
+        build_failed = True
+        with open("./logs/failed_build.txt", "a+") as f:
+            f.write(f"{build_name}|Check logs\n")
+    # check testsuite failed
+    artifact_name += "-report.log"
+    artifact = repo.get_artifacts(artifact_name).get_page(0)
+    if len(artifact) == 0:
+        print(f"testsuite failed for {build_name}")
+        if not build_failed:
+            with open("./logs/failed_testsuite.txt", "a+") as f:
+                f.write(f"{build_name}|Cannot find testsuite artifact\n")
+        return -1
+    return artifact[0].id
+
+def download_artifact(artifact_name: str, artifact_id: str, token: str):
+    """
+    Uses GitHub api endpoint to download and extract the log artifact into
+    directory called ./logs
+    """
+    params = {"Accept": "application/vnd.github+json",
+              "Authorization": f"token {token}",
+              "X-Github-Api-Version": "2022-11-28"}
+    artifact_zip_name = artifact_name.replace(".log", ".zip")
+    print(artifact_zip_name)
+    print(f"downloading: {artifact_id}")
+    r = requests.get(f"https://api.github.com/repos/patrick-rivos/riscv-gnu-toolchain/actions/artifacts/{artifact_id}/zip", headers=params)
+    print(r.status_code)
+    with open(f"./{artifact_zip_name}", "wb") as f:
+        f.write(r.content)
+    with ZipFile(f"./{artifact_zip_name}", "r") as zf:
+        zf.extractall(path=f"./{artifact_name.split('.log')[0]}")
+    os.rename(f"./{artifact_name.split('.log')[0]}/{artifact_name}", f"./logs/{artifact_name}")
+
+def download_all_artifacts(current_hash: str, token: str):
+    """
+    Goes through all possible artifact targets and downloads it
+    if it exists. Downloads previous successful hash's artifact
+    as well. Runs comparison on the downloaded artifacts
+    """
+
+    auth = Auth.Token(token)
+    g = Github(auth=auth)
+
+    repo = g.get_repo("patrick-rivos/riscv-gnu-toolchain")
+
+    prev_commits = gcc_hashes(current_hash, False)
+    artifact_names = get_possible_artifact_names()
+    for artifact_name in artifact_names:
+        artifact = artifact_name.format(current_hash)
+        artifact_id = check_artifact_exists(artifact, repo)
+        if artifact_id == -1:
+            continue
+        # comparison output path
+        compare_path = f"./summaries/{artifact + '-report-summary.md'}"
+        artifact += "-report.log"
+        artifact_name += "-report.log"
+        # download current artifact
+        download_artifact(artifact, str(artifact_id), token)
+
+        # download previous artifact
```

Comment:
It seems like we're trying to run without a baseline and then download the previous artifact? I think it would make more sense when reading to try downloading the artifact and if it's invalid then run without the baseline. (wrap it all in an if/else)

Reply:
I'm a bit confused as to what you mean by trying to run it without a baseline. `check_artifact_exists` returns `-1` if the desired artifact does not exist and skip that artifact with the `continue` statement. Therefore, we know that the current hash artifact always exists before downloading. After we download the current hash's artifact, we check to see if the baseline exists with `get_valid_artifact_hash` and checking for `base_hash == "No valid hash"`

Reply:
Got it. The structure of the check/`continue` threw me off a bit :)